test/junit/scala/collection/immutable/LazyListLazinessTest.scala

```diff
@@ -811,14 +811,12 @@ class LazyListLazinessTest {
 
     val tailInitFactory = lazyListFactory { init =>
       def gen(index: Int): LazyList[Int] = {
+        def elem(): LazyList[Int] = { init.evaluate(index); LazyList.fill(1)(index) }
         if (index >= LazinessChecker.count) LazyList.empty
-        else {
-          init.evaluate(index)
-          LazyList.fill(1)(index) #::: gen(index + 1)
-        }
+        else elem() #::: gen(index + 1)
```

Comment:
I thought this was maybe cleaner, but am happy to revert it if others disagree. it's just rearranging things a bit, not changing logic